webapp/channels/src/components/user_settings/modal/user_settings_modal.tsx

```diff
@@ -317,60 +322,58 @@ class UserSettingsModal extends React.PureComponent<Props, State> {
         const {formatMessage} = this.props.intl;
 
         let modalTitle: string;
-
         if (this.props.adminMode && this.props.user) {
-            modalTitle = formatMessage({
-                id: 'userSettings.adminMode.modal_header',
-                defaultMessage: "{userDisplayName}'s Settings",
-            }, {
-                userDisplayName: getDisplayName(this.props.user),
-            });
+            modalTitle = formatMessage(
+                {id: 'userSettings.adminMode.modal_header', defaultMessage: "{userDisplayName}'s Settings"},
+                {userDisplayName: getDisplayName(this.props.user)},
+            );
+        } else if (this.props.isContentProductSettings) {
+            modalTitle = formatMessage({id: 'global_header.productSettings', defaultMessage: 'Settings'});
         } else {
-            modalTitle = this.props.isContentProductSettings ? formatMessage({
-                id: 'global_header.productSettings',
-                defaultMessage: 'Settings',
-            }) : formatMessage({
-                id: 'user.settings.modal.title',
-                defaultMessage: 'Profile',
-            });
+            modalTitle = formatMessage({id: 'user.settings.modal.title', defaultMessage: 'Profile'});
         }
 
+        const headerTitle = (
+            <div
+                className='UserSettingsModal__headerWrapper'
+                id='accountSettingsHeader'
+            >
+                <span
+                    id='accountSettingsModalLabel'
+                    className='modal-header__title'
+                >
+                    {modalTitle}
+                </span>
+                {this.props.adminMode && (
+                    <div className='adminModeBadge'>
+                        <FormattedMessage
+                            id='userSettings.adminMode.admin_mode_badge'
+                            defaultMessage='Admin Mode'
+                        />
+                    </div>
+                )}
+            </div>
+        );
+
         return (
-            <Modal
+            <GenericModal
                 id='accountSettingsModal'
-                dialogClassName='a11y__modal settings-modal'
+                className='a11y__modal settings-modal UserSettingsModal'
                 show={this.state.show}
                 onHide={this.handleHide}
                 onExited={this.handleHidden}
+                ariaLabel={modalTitle}
+                compassDesign={true}
                 enforceFocus={this.state.enforceFocus}
-                aria-label={modalTitle}
-                role='none'
+                bodyPadding={false}
+                modalHeaderText={headerTitle}
             >
-                <Modal.Header
-                    id='accountSettingsHeader'
-                    closeButton={true}
+                <div
+                    ref={this.modalBodyRef}
+                    className='UserSettingsModal__bodyWrapper'
```

Comment:
Why add this div wrapping everything? I don't see it referenced in the CSS at all